api/python/quilt3/admin/__init__.py

```diff
@@ -0,0 +1,178 @@
+"""
+APIs for Quilt administrators. 'Registry' refers to Quilt stack backend services, including identity management.
+"""
+# This wraps code generated by aridne-codegen to provide a more user-friendly API.
+from datetime import datetime
+from typing import Annotated, Any, List, Literal, Optional, Union
+
+import pydantic
+
+from . import _graphql_client
+
+
+@pydantic.dataclasses.dataclass
+class ManagedRole:
+    id: str
+    name: str
+    arn: str
+    typename__: Literal["ManagedRole"]
+
+
+
+@pydantic.dataclasses.dataclass
+class UnmanagedRole:
+    id: str
+    name: str
+    arn: str
+    typename__: Literal["UnmanagedRole"]
+
+
+Role = Union[ManagedRole, UnmanagedRole]
+AnnotatedRole = Annotated[Role, pydantic.Field(discriminator="typename__")]
+role_adapter = pydantic.TypeAdapter(AnnotatedRole)
+
+
+@pydantic.dataclasses.dataclass
+class User:
+    name: str
+    email: str
+    date_joined: datetime
+    last_login: datetime
+    is_active: bool
+    is_admin: bool
+    is_sso_only: bool
+    is_service: bool
+    role: Optional[AnnotatedRole]
+    extra_roles: List[AnnotatedRole]
+
+
+class Quilt3AdminError(Exception):
+    def __init__(self, details):
+        super().__init__(details)
+        self.details = details
+
+
+class UserNotFoundError(Quilt3AdminError):
+    def __init__(self):
+        super().__init__(None)
+
+
+def _handle_errors(result: _graphql_client.BaseModel) -> Any:
+    if isinstance(result, (_graphql_client.InvalidInputSelection, _graphql_client.OperationErrorSelection)):
+        raise Quilt3AdminError(result)
+    return result
+
+
+def _get_client():
+    return _graphql_client.Client()
+
+
+def get_user(name: str) -> Optional[User]:
+    """
+    Get a specific user from the registry. Return `None` if the user does not exist.
+
+    Args:
+        name: Username of user to get.
+    """
+    result = _get_client().get_user(name=name)
+    if result is None:
+        return None
+    return User(**result.model_dump())
```

Comment:
i'd prefer making this a classmethod on User, tho not sure it's worth it

Reply:
yes, and `User.set_admin()` and `User.set_email()` and so on

Reply:
i'm not against that in principle, but it would require thorough design considerations.
this kind of "resource"-oriented api feels kinda more complex, bc you have to distinguish between entity identity vs attributes/props.
e.g. do you have to fetch the user before mutating it, like `User.get(name).set_role(role)`?
or can you just construct a "resource object" by providing an identifier, e.g. `User(name).set_role(role)`?

Reply:
> e.g. do you have to fetch the user before mutating it, like User.get(name).set_role(role)?

this

Reply:
> i'm not against that in principle, but it would require thorough design considerations.
this kind of "resource"-oriented api feels kinda more complex, bc you have to distinguish between entity identity vs attributes/props.

yes, so I'd probably leave things as is and we could rework it in future